### DISCUSSION TITLE
Object interaction

### DISCUSSION THREAD

**Discussion by ChemenInfiskill** [flareteam/flare-game#951] [Q&A]
Is there a way to tie an event to a player action ? I'd like for doors to open only when clicked on by the player, for example. With a common event, you just have to move to a specific location for the event to be triggered. I'd like the event to be triggered only when the player consciously performs a certain action, like clicking or pressing a key on the related object / location.

I searched some of the game maps for a way to do this, but could not find an example.

Thanks in advance for any help.

**Comment by dorkster** [comment]
The only way to accomplish this is to place collision tiles on the event to prevent the player from standing on it. In that case, the event can only be triggered by clicking on it or by pressing the "Accept" key when in range of the event.

**EDIT**: I forgot we also have `on_interact` for `event.activate`, which does what I described above without needing to block the collision.

**Reply by ChemenInfiskill** [reply]
Thanks for the tip. There is no way to trigger the event using a range weapon like an arrow, fireball or such ?

**Reply by dorkster** [reply]
No, hazards can only affect entities (player, enemies, allies).

If you want an event that can *only* be triggered by an attack, you can get a bit creative and have an "enemy" with a power that triggers `on_death`. Then said power can call a `script` file.

**Reply by ChemenInfiskill** [reply]
Ok. Thank you. I thought of a similar workaround, but wanted to be sure there was no better way to handle the event.

Thanks for the prompt response.
